src/twister2/twister_config.py

```diff
@@ -27,6 +27,11 @@ class TwisterConfig:
     fixtures: list[str] = field(default_factory=list, repr=False)
     extra_args_cli: list = field(default_factory=list)
     overflow_as_errors: bool = False
+    integration_mode: bool = False
+    emulation_only: bool = False
+    architectures: list[str] = field(default_factory=list, repr=False)
+    default_platforms: bool = False
+    platform_filter: list[str] = field(default_factory=list, repr=False)
```

Comment:
Maybe to be more precisely this variable should have name like: `platform_filter_cli` to highlight, that this filter came from users CLI call. At this moment at first glance it can be hard to understand what is difference between `selected_platfroms` and `platform_filter`.

Reply:
In `platform_filter` can be also stored platforms from `hardware_map` if no `--platform` was given